src/hazelcore/dynamics/Elaborator_Exp.re

```diff
@@ -529,17 +529,16 @@ and syn_elab_line =
     }
   | EmptyLine
   | CommentLine(_) => LinesExpand(d => d, ctx, delta)
-  | LetLine(p, ann, def) =>
-    switch (ann) {
-    | Some(uty1) =>
-      let ty1 = UHTyp.expand(uty1);
-      let (ctx1, is_recursive_fn) =
-        Statics_Exp.ctx_for_let(ctx, p, ty1, def);
+  | LetLine(p, def) =>
```

Comment:
Coming across this code as I merge dev into livelits. Strikes me as an early change that was never evolved with later realizations about desired behavior. I'd expect the pattern to be elaborated by `Elaborator_Pat.syn_elab` rather than `ana_elab` further below in unchanged lines; also weird that the input to this `ana_elab` call is `ty1` which was synthesized from the pattern.

Not a blocker for me so no rush.